From 895e76e75dbb993a8f445072c190a9db4ee50d15 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Sat, 7 Mar 2020 08:34:02 -0800
Subject: [PATCH] Fix nss multilib build on openSUSE 11.x 32bit

While building lib64-nss on openSUSE 11.x 32bit, the nsinstall will
fail with error:

* nsinstall.c:1:0: sorry, unimplemented: 64-bit mode not compiled

It caused by the '-m64' option which passed to host gcc.

The nsinstall was built first while nss starting to build, it only runs
on host to install built files, it doesn't need any cross-compling or
multilib build options. Just clean the ARCHFLAG and LDFLAGS to fix this
error.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 nss/coreconf/nsinstall/Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/nss/coreconf/nsinstall/Makefile b/nss/coreconf/nsinstall/Makefile
index 337cbeb..dd77fbe 100644
--- a/nss/coreconf/nsinstall/Makefile
+++ b/nss/coreconf/nsinstall/Makefile
@@ -18,6 +18,13 @@ INTERNAL_TOOLS  = 1
 
 include $(DEPTH)/coreconf/config.mk
 
+# nsinstall is unfit for cross-compiling/multilib-build since it was
+# always run on local host to install built files. This change intends
+# to clean the '-m64' from ARCHFLAG and LDFLAGS.
+ARCHFLAG =
+LDFLAGS =
+# CFLAGS =
+
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 PROGRAM		=
 TARGETS		=
-- 
2.25.1

